watch-parties/src/main/webapp/chatscript.js

```diff
@@ -1,21 +1,39 @@
-//Timer under construction
-//document.getElementById("chatlog").onload = function() {setInterval(window.alert("hi"), 5000)}; 
-document.getElementById("sendButton").onclick = function() {addNewComment()}; 
+document.getElementById("sendButton").addEventListener("click", addNewComment);
 var formattedComment = "";
+var timer = window.setInterval(loadComments, 1000); //check comment updates every second
 
+// retrieves comments from the backend, loads using AJAX
+function loadComments() {
+  var xhttp;
+  xhttp = new XMLHttpRequest();
+  xhttp.onreadystatechange = function() {
+    if (this.readyState == 4 && this.status == 200) {
+    document.getElementById("chatlog").innerHTML = this.responseText;
+    }
+  };
```

Comment:
You can just listen for when the request is complete instead of every time its ready state updates.
https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/load_event
```suggestion
  xhttp.addEventListener('load', function() {
    if (this.status == 200) {
      document.getElementById("chatlog").innerHTML = this.responseText;
    }
  });
```